common/io/io.cpp

```diff
@@ -92,50 +92,93 @@ Error Copy(Writer &dst, Reader &src, vector<uint8_t> &buffer) {
 	}
 }
 
-void AsyncCopy(Writer &dst, AsyncReader &src, function<void(Error)> finished_handler) {
+struct CopyData {
+	CopyData(size_t limit) :
+		buf(MENDER_BUFSIZE),
+		limit(limit) {
```

Comment:
I'd use `{}` here since it's used for `copied` below, but YMMV. And especially for the vectors I still find it quite confusing, so...

Reply:
Done!

And indeed it is confusing for vectors, but doing it like this both expresses the correct intent, and does the correct thing too:

```c++
CopyData(size_t limit) :
		buf(MENDER_BUFSIZE),
		limit {limit} {
```

* `buf` constructed using a function, since it is not the value we are assigning here, but the size.
* `limit` constructed using brace-init, since we are assigning a value.